src/main.rs

```diff
@@ -29,18 +33,32 @@ fn main() {
         .init();
     log::info!("Starting up...");
 
+    // Read a file name as the CLI argument
+    let (file_path, initial_json) = if let Some(first_arg) = std::env::args().skip(1).next() {
+        let path = PathBuf::from(first_arg);
+        let file = match std::fs::File::open(&path) {
+            Ok(x) => x,
+            Err(e) => {
+                eprintln!("Error opening {:?}: {}", path, e);
+                return;
+            }
+        };
+        (
+            Some(path),
+            serde_json::from_reader::<_, Value>(file).unwrap(),
+        )
+    } else {
+        log::warn!("Expected a file-name as an argument.  Using default JSON instead.");
+        (None, serde_json::json!([true, false, { "value": false }]))
+    };
+
```

Comment:
@Kixunil I think I've de-janked all the file-reading code as much as possible (as well as printing the `PathBuf` along with the errors).  But I'm no expert on idiomatic Rust, so you're welcome to further idiomaticise it :grin:.

Reply:
This is good for starters, I was previously thinking about making an error type. (We will likely need it too when we implement `:e`.)

Reply:
Cool - I'll get this merged to avoid merge conflicts with other PRs.